crates/rolldown/src/build/build.rs

```diff
@@ -0,0 +1,345 @@
+use crate::build::build_context::BuildContext;
+
+use super::super::{
+  SharedOptions, SharedResolver,
+  bundler::CacheGuard,
+  module_loader::deferred_scan_data::defer_sync_scan_data,
+  stages::{
+    generate_stage::GenerateStage,
+    link_stage::LinkStage,
+    scan_stage::{NormalizedScanStageOutput, ScanStage, ScanStageOutput},
+  },
+  types::{bundle_output::BundleOutput, scan_stage_cache::ScanStageCache},
+  utils::fs_utils::clean_dir,
+};
+use arcstr::ArcStr;
+use rolldown_common::{GetLocalDbMut, Module, ScanMode, SharedFileEmitter, SymbolRefDb};
+use rolldown_debug::{action, trace_action, trace_action_enabled};
+use rolldown_error::{BuildDiagnostic, BuildResult, Severity};
+use rolldown_fs::{FileSystem, OsFileSystem};
+use rolldown_plugin::{HookBuildEndArgs, HookRenderErrorArgs, SharedPluginDriver};
+use rolldown_utils::dashmap::FxDashSet;
+use std::sync::Arc;
+
+pub struct Build {
+  pub(crate) fs: OsFileSystem,
+  pub(crate) options: SharedOptions,
+  pub(crate) resolver: SharedResolver,
+  pub(crate) file_emitter: SharedFileEmitter,
+  pub(crate) plugin_driver: SharedPluginDriver,
+  pub(crate) warnings: Vec<BuildDiagnostic>,
+  pub(crate) cache: ScanStageCache,
+  pub(crate) session: rolldown_debug::Session,
+}
+
+impl Build {
+  #[tracing::instrument(level = "debug", skip_all, parent = &self.session.span)]
+  /// This method intentionally to get the ownership of `self` to show that the method can be called multiple times.
```

Comment:
```suggestion
  /// This method intentionally get the ownership of `self` to show that the method cannot be called multiple times.
```
(other places as well)